src/plugins/discover/public/components/common/error_alert.tsx

```diff
@@ -15,8 +15,9 @@ import { useDiscoverServices } from '../../hooks/use_discover_services';
 export const DiscoverError = ({ error }: { error: Error }) => {
   const { locator } = useDiscoverServices();
 
-  const goToMain = () => {
-    locator.navigate({});
+  const goToMain = async () => {
+    await locator.navigate({});
+    window.location.reload();
```

Comment:
This is for the case when a saved search Id (from URL) is not found. It's not resetting the error state unless reloaded.

Reply:
We should aim to reset the error state higher up the component tree, I think 
adding ` setError(undefined);` before/after 
https://github.com/elastic/kibana/blob/2e80281d5fcedd1d71121bbae80511e573fa47da/src/plugins/discover/public/application/main/discover_main_route.tsx#L135C1-L135C1

should do the trick 

not for this PR, but I think there's plenty of potential to simplify our `useState` usage here

```
  const [error, setError] = useState<Error>();
  const [loading, setLoading] = useState(true);
  const [hasESData, setHasESData] = useState(false);
  const [hasUserDataView, setHasUserDataView] = useState(false);
  const [showNoDataPage, setShowNoDataPage] = useState<boolean>(false);
```

Reply:
thx for applying this feedback, resolving conversation